packages/wallet-ui/src/utils/utils.ts

```diff
@@ -45,22 +50,31 @@ export const isValidAddress = (toCheck: string) => {
 export const addMissingPropertiesToToken = (
   token: Erc20Token,
   balance?: string,
+  balanceSpendable?: string,
   usdPrice?: number,
 ): Erc20TokenBalance => {
   return {
     ...token,
     amount: ethers.BigNumber.from(balance ? balance : '0x0'),
+    spendableAmount: ethers.BigNumber.from(
+      balanceSpendable ? balanceSpendable : '0x0',
+    ),
     usdPrice: usdPrice,
   };
 };
 
 export const getHumanReadableAmount = (
   asset: Erc20TokenBalance,
   assetAmount?: string,
+  balanceType?: BalanceType,
 ) => {
+  const amount =
+    balanceType === BalanceType.Spendable
+      ? asset.spendableAmount ?? asset.amount
+      : asset.amount;
```

Comment:
i think this method has been used in some place, and mainly purpose is to format the given value, 
assetAmount is a var to override the assert val if needed

e.g when a tx history display, it call this method and override the token val (balance) by the tx val

so i think may be not good to add a logic here to see if it is spendable, 

do you think it is better to put those logic from caller?

Reply:
agreed, moved up.